sentry/src/main/java/io/sentry/SendCachedEnvelopeFireAndForgetIntegration.java

```diff
@@ -112,6 +113,17 @@ private synchronized void sendCachedEnvelopes(@NotNull IHub hub, @NotNull Sentry
       return;
     }
 
+    // in case there's rate limiting active, skip processing
```

Comment:
could we still run into problems, where we're not rate limited before starting to send a ton of events, then halfway through we get rate limited and still run into problems? Should this (also) be checked in or around e.g. `processFile`?

Reply:
yeah I think it's a good idea to check inside the for-loop 👍